online_meetings/report.py

```diff
@@ -83,31 +79,47 @@ def get_total_page_count(url: str, headers: Dict[str, Union[str, int]], params:
     return total_page_count
 
 
-def run_report(api_url: str, headers: Dict[str, Union[str, int]], json_attribute_name: str,
+def run_report(api_url: str, json_attribute_name: str,
                default_params: Dict[str, Union[str, int]] = None, page_size: int = 300,
                page_token: bool = False, use_date: bool = False):
     if default_params is None:
         default_params = {}
-    url = ZOOM_BASE_URL + api_url
     params = default_params
     # If page size is specified use this
     if page_size:
         params['page_size'] = page_size
     total_list = []
     # TODO: Detect the date from the previous CSV
     # Either loop for all dates or just one a single report
-    if use_date:
-        early_date = dateparser.parse(ENV.get('ZOOM_EARLIEST_FROM', '2020-03-01')).date()
-        # Only use the date as a paramter
-        # Loop until yesterday (this will go until now() -1)
-        for i in range((datetime.now().date() - early_date).days):
-            param_date = early_date + timedelta(days=i)
-            params["from"] = str(param_date)
-            params["to"] = str(param_date)
-            # Add this loop to the list
-            total_list.extend(zoom_loop(url, headers, json_attribute_name, dict(params), page_token))
-    else:
-        total_list.extend(zoom_loop(url, headers, json_attribute_name, dict(params), page_token))
+
+    for zoom_key, zoom_config in ENV["ZOOM_CONFIG"].items():
+        logger.info(f"Starting zoom pull for instance {zoom_key}")
+        url = zoom_config["BASE_URL"] + api_url
+        token = jwt.encode(
+            {'iss': zoom_config["API_KEY"],
+             'exp': datetime.utcnow() + timedelta(minutes=30)
```

Comment:
Assuming I'm interpreting this correctly, would we ever need more than 30 minutes for an account run?

Reply:
Yeah that's a good point. I think we either need a token refresh function or a longer time. Currently it's taking 20 minutes and if it went beyond that this would just stop running. I'll look at that as it might be more useful in the future too.

Reply:
So, do each of them take 20 minutes?

Reply:
No, the whole process is. I don't have that timed down to the specific run. The second zoom is like 10% less than the first so I'd guess it's 10% of the total.

Reply:
I don't think the token refresh will be hard to implement. We can even pro-actively do it without getting expired response back from the server. Though really I think the expiration time could also just be set to be much longer and Zoom doesn't care. Like we've had a JWT token that's had over a week expiration.

Reply:
Okay I rewrote this entire method to use a new class ZoomJWT. This when passed to requests will take care of setting the header and also doing the refresh if needed. This is now done as close as possible to the actual calls and all even more easily swappable.